### PR TITLE
Easier-to-review version of #33

### PR DESCRIPTION
**Pull request by jamalex**
This includes only the changes I made on top of @jayoshih's work, making it easier to review.

This targets a dummy branch, so this shouldn't be merged directly. Instead:

1. Review this PR, and we can iterate here
2. Once it's good to go, merge #33, and close this PR

### CODE REVIEW

File: ricecooker/__main__.py [hunk 1]
```diff
@@ -33,10 +33,10 @@
 
 """
 
-from ricecooker.commands import uploadchannel
-from ricecooker import config
-from ricecooker.exceptions import InvalidUsageException
-from ricecooker.managers import Status
+from .commands import uploadchannel
```
Comment (jamalex) [inline ricecooker/__main__.py]: Relativized imports to get things working properly in dev mode.

File: ricecooker/classes/nodes.py [hunk 2]
```diff
@@ -150,7 +153,7 @@ def validate(self):
             assert isinstance(self.thumbnail, str) or self.thumbnail is None
             return super(Channel, self).validate()
         except AssertionError as ae:
-            raise InvalidNodeException("Invalid node: {0} - {1}".format(self.title, ae))
+            raise InvalidNodeException("Invalid channel ({}): {} - {}".format(ae.args[0], self.title, self.__dict__))
```
Comment (jamalex) [inline ricecooker/classes/nodes.py]: Added the assert error message to these to provide more information (see https://github.com/learningequality/ricecooker/issues/26)

File: ricecooker/classes/nodes.py [hunk 3]
```diff
@@ -530,4 +533,61 @@ def validate(self):
 
             return super(Exercise, self).validate()
         except AssertionError as ae:
-            raise InvalidNodeException("Invalid node: {0} - {1}".format(self.title, self.__dict__))
+            raise InvalidNodeException("Invalid node ({}): {} - {}".format(ae.args[0], self.title, self.__dict__))
+
+
```
Comment (jamalex) [inline ricecooker/classes/nodes.py]: Section below is the important bit.

File: ricecooker/classes/questions.py [hunk 4]
```diff
@@ -238,6 +236,38 @@ def process_image_field(self, data, downloader):
             new_data[new_key] = new_data.pop(k)
         return new_data, files
```
Comment (jamalex) [inline ricecooker/classes/questions.py]: May be able to delete the method below; might be residual from one of the many merge conflicts, and no longer needed.

File: ricecooker/commands.py [hunk 5]
```diff
@@ -3,10 +3,17 @@
...
 from requests.exceptions import HTTPError
-from ricecooker.managers import ChannelManager, RestoreManager, Status
+from .managers import ChannelManager, RestoreManager, Status
+
+# Fix to support Python 2.x.
+# http://stackoverflow.com/questions/954834/how-do-i-use-raw-input-in-python-3
+try:
+    input = raw_input
+except NameError:
+    pass
```
Comment (jamalex) [inline ricecooker/commands.py]: I hadn't realized at the time I added this that there was a decision to give up on 2.x support. If we don't plan to try to support 2.x at all, I can remove this.

File: ricecooker/classes/questions.py (hunk 4 above)
Comment (jayoshih) [inline ricecooker/classes/questions.py]: Will be used in a future branch

**Pull request closed without merging**